### PR TITLE
refactor sox_cldaero_mod

### PR DESCRIPTION
**Pull request by tangshuaiqi**
include refactor of subroutine `sox_cldaero_create_obj` and `sox_cldaero_update:`

- remove unused input variables and code that are not related to MAM4 (such as `msa`, `no3`, etc)
- add descriptions and comments to better understand input/output and local variables
- separate subroutines `compute_aer_factor,` `update_tmr,` `update_vmr_nonzero,` `calc_sfc_flux`
- reorganize a few code, rename single-digit variables

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90 [hunk 1]
```diff
@@ -88,436 +87,328 @@ subroutine sox_cldaero_init
...
-  end subroutine sox_cldaero_update
+    ! at this point (sumf <= 0.0) only when all the faqgain_so4 are zero
+    if (sumf > 0.0_r8) faqgain_so4(:) = faqgain_so4(:)/sumf
+
+  end subroutine compute_aer_factor
+
+  !=============================================================================
+  subroutine update_tmr ( tmr, dqdt, dtime )
+    !-----------------------------------------------------------------------
+    ! basically it just makes sure the value is greater than zero
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90]: This isn't correct, is it? This updates `tmr` by applying the tendencies over the time interval `dt`.

Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90]: aha, sorry. This was just copy-pasted from the subroutine update_tmr_nonzero below... will change it.

File: components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90 [hunk 2]
```diff
@@ -88,436 +87,328 @@ subroutine sox_cldaero_init
...
+          cloud: if ((cldfrc(icol,kk) >= small_value_5) .and. (xlwc(icol,kk) >= small_value_8)) then
+
+            !-------------------------------------------------------------------------
+            ! compute factors for partitioning aerosol mass gains among modes
+            ! the factors are proportional to the activated particle MR for each
+            ! mode, which is the MR of cloud drops "associated with" the mode
+            ! thus we are assuming the cloud drop size is independent of the
+            ! associated aerosol mode properties (i.e., drops associated with
+            ! Aitken and coarse sea-salt particles are same size)
+            call  compute_aer_factor(qcw(icol,kk,:), loffset, faqgain_so4)
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/sox_cldaero_mod.F90]: Mention that `faqgain_so4` is an output.

**Review by singhbalwinder — APPROVED** [review]
Great job on this! I have left a very minor comment.

**Comment by tangshuaiqi** [comment]
All tests passed, merge this branch.

comp_testall_sox_cldaero_mam4_org_v2_baselines: 9 tests
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P1x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**